Sources/SwiftProtobuf/FieldNameMap.swift

```diff
@@ -8,6 +8,82 @@
 //
 // -----------------------------------------------------------------------------
 
+/// Proto field names are always ASCII, so we can improve the
+/// performance of some of our parsing by storing them as
+/// arrays of UInt8 to avoid the overhead of creating and
+/// comparing Unicode Strings.
```

Comment:
This feels a bit rough and tricky to maintain going forward—I think we can improve this quite a bit.

There are two situations where we need to create instances of these field names: in the generated code (where it's a literal string), and while parsing JSON/text input (where we have a UBP<UInt8>). In both cases, we know they should be ASCII.

So, we could design our own small `ASCIIString` type as follows:
* It internally holds an `UnsafeBufferPointer<UInt8>` to the string contents
* It conforms to `ExpressibleByStringLiteral` using `StaticString` as its `StringLiteralType`. This handles the case for the names in generated code; since you can access the `UBP<UInt8>` of a `StaticString` directly, and since static strings are stored in the data segment, we don't have to worry about retention.
* It has another `init` that takes a `UBP<UInt8>` directly, which handles the case of names scanned during parsing. As long as we ensure that an instance created this way doesn't escape its calling context, we're good.
* It conforms to `Hashable` and `Equatable` for the dictionary key use case. (`Equatable` can just be implemented in terms of `UBP<UInt8>.elementsEqual` instead of hand-rolling a loop.)

A side effect of this is that `ASCIIString` will have to be public since it will become the `FieldNameMaps.Names` payload type instead of `String`.

This should give you the same performance benefits but will be easier to work with in the future. (Indeed, the performance might be a bit better because you're eliminating all the `String` usage, including the ones that get created when the dictionary is first initialized. So there should be some heap memory usage improvements here too.)